CRM/Report/Form/Contribute/DeferredRevenue.php

```diff
@@ -65,123 +247,93 @@ public function preProcess() {
-LEFT JOIN  civicrm_membership membership 
+  ON line_item.contribution_id = {$this->_aliases['civicrm_contribution']}.id
+LEFT JOIN  civicrm_membership {$this->_aliases['civicrm_membership']}
   ON CASE
     WHEN line_item.entity_table = 'civicrm_membership'
-    THEN line_item.entity_id = membership.id
-    ELSE membership.id = 0
+    THEN line_item.entity_id = {$this->_aliases['civicrm_membership']}.id
+    ELSE {$this->_aliases['civicrm_membership']}.id = 0
   END
-LEFT JOIN civicrm_participant participant
+LEFT JOIN civicrm_participant {$this->_aliases['civicrm_participant']}
   ON CASE
     WHEN line_item.entity_table = 'civicrm_participant'
-    THEN line_item.entity_id = participant.id
-    ELSE participant.id = 0
+    THEN line_item.entity_id = {$this->_aliases['civicrm_participant']}.id
+    ELSE {$this->_aliases['civicrm_participant']}.id = 0
   END
-LEFT JOIN civicrm_event event ON participant.event_id = event.id
+LEFT JOIN civicrm_event {$this->_aliases['civicrm_event']} ON {$this->_aliases['civicrm_participant']}.event_id = {$this->_aliases['civicrm_event']}.id
 ";
-  }
-
-  public function orderBy() {
-    parent::orderBy();
-  }
-
-  public function where() {
-    $clauses = array();
-    foreach ($this->_columns as $tableName => $table) {
-      if (array_key_exists('filters', $table)) {
-        foreach ($table['filters'] as $fieldName => $field) {
-          $clause = NULL;
-          if (CRM_Utils_Array::value('type', $field) & CRM_Utils_Type::T_DATE) {
-            $relative = CRM_Utils_Array::value("{$fieldName}_relative", $this->_params);
-            $from = CRM_Utils_Array::value("{$fieldName}_from", $this->_params);
-            $to = CRM_Utils_Array::value("{$fieldName}_to", $this->_params);
 
-            $clause = $this->dateClause($field['name'], $relative, $from, $to, $field['type']);
-          }
-          else {
-            $op = CRM_Utils_Array::value("{$fieldName}_op", $this->_params);
-            if ($op) {
-              $clause = $this->whereClause($field,
-                $op,
-                CRM_Utils_Array::value("{$fieldName}_value", $this->_params),
-                CRM_Utils_Array::value("{$fieldName}_min", $this->_params),
-                CRM_Utils_Array::value("{$fieldName}_max", $this->_params)
-              );
-            }
-          }
-          if (!empty($clause)) {
-            $clauses[] = $clause;
-          }
-        }
-      }
-    }
-    if (!empty($clauses)) {
-      $this->_where = 'WHERE ' . implode(' AND ', $clauses);
+    if (!empty($this->_params['batch_id_value'])) {
+      $this->_from .= "
+        LEFT JOIN civicrm_entity_batch {$this->_aliases['civicrm_batch']}
+          ON {$this->_aliases['civicrm_batch']}.entity_id = {$this->_aliases['civicrm_financial_trxn_1']}.id AND
+            {$this->_aliases['civicrm_batch']}.entity_table = 'civicrm_financial_trxn'\n";
     }
   }
 
   public function postProcess() {
     $this->_noFields = TRUE;
-    parent::postProcess();
```

Comment:
Why is this function being overridden - if there is a reason then add a comment to explain. It might be cleaner just to override the limit() function if that is the reason